java/code/src/com/suse/manager/webui/controllers/DownloadController.java

```diff
@@ -378,11 +376,13 @@ public static Object downloadPackage(Request request, Response response) {
         String channel = request.params(":channel");
         String path = "";
         try {
-            URL url = new URL(request.url());
+            URI uri = new URI(request.url());
             // URL decode the path to support ^ in package versions
-            path = URLDecoder.decode(url.getPath(), StandardCharsets.UTF_8);
+            path = uri.getPath();
+
         }
-        catch (MalformedURLException e) {
+        catch (URISyntaxException e) {
+            log.error(String.format("Unable to parse: %s", request.url()));
```

Comment:
```suggestion
            log.error("Unable to parse: {}", request.url());
```
Here and below, no need to use `String.format` anymore. Log4j 2 supports formatting out of the box.

Reply:
Ok, this this cannot be backported to 4.2 as it still uses lib4j 1.x